### PR TITLE
[Merged by Bors] - feat: Functor.mapHomotopyCategory is a triangulated functor

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we show that if `F` is an additive functor, then the induced functor on the homotopy categories is a triangulated functor.

---

- [x] depends on: #14087

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/HomotopyCategory/MappingCone.lean [hunk 1]
```diff
@@ -541,6 +541,93 @@ lemma lift_desc_f {K L : CochainComplex C ℤ} (α : Cocycle K F 1) (β : Cochai
...
 
+
+section
+
+open Preadditive Category
+
+variable (H : C ⥤ D) [H.Additive]
+  [HasHomotopyCofiber ((H.mapHomologicalComplex (ComplexShape.up ℤ)).map φ)]
+
+/-- If `H : C ⥤ D` is an additive functor and `φ` is a morphism of cochain complexes
```
Comment (semorrison) [inline Mathlib/Algebra/Homology/HomotopyCategory/MappingCone.lean:552]: Could the doc string (or a simple comment) explain why this is here separately from the unprimed version, and presumably direct the reader's attention to that?

File: Mathlib/Algebra/Homology/HomotopyCategory/MappingCone.lean [hunk 2]
```diff
@@ -541,6 +541,93 @@ lemma lift_desc_f {K L : CochainComplex C ℤ} (α : Cocycle K F 1) (β : Cochai
...
+@[simps]
+noncomputable def mapHomologicalComplexXIso' (n m : ℤ) (hnm : n + 1 = m) :
+    ((H.mapHomologicalComplex (ComplexShape.up ℤ)).obj (mappingCone φ)).X n ≅
+      (mappingCone ((H.mapHomologicalComplex (ComplexShape.up ℤ)).map φ)).X n where
+  hom := H.map ((fst φ).1.v n m (by omega)) ≫
+      (inl ((H.mapHomologicalComplex (ComplexShape.up ℤ)).map φ)).v m n (by omega) +
+      H.map ((snd φ).v n n (add_zero n)) ≫
+        (inr ((H.mapHomologicalComplex (ComplexShape.up ℤ)).map φ)).f n
+  inv := (fst ((H.mapHomologicalComplex (ComplexShape.up ℤ)).map φ)).1.v n m (by omega) ≫
+      H.map ((inl φ).v m n (by linarith)) +
```
Comment (semorrison) [inline Mathlib/Algebra/Homology/HomotopyCategory/MappingCone.lean:564]: Not at a computer so I can't check: could this be `omega`? No problem if there is a good reason.

**Comment by semorrison** [comment]
bors d+

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**